src/haddock/libs/libplots.py

```diff
@@ -107,10 +112,19 @@ def box_plots(capri_filename, cl_ranking):
         else:
             cl_df["capri_rank"] = cl_ranking[cl_id]
             gb_good = pd.concat([gb_good, cl_df])
+            
     gb_other["cluster-id"] = "Other"
     gb_other["capri_rank"] = len(cl_ranking.keys()) + 1
     gb_cluster = pd.concat([gb_good, gb_other])
+    # when png is true we already order everything according to capri_rank
+    if png:
```

Comment:
Why do you need this check here?

Reply:
it's a technicality: plotly can act directly on pandas dataframe, sorting them according to whatever order (column). For matplotlib that's not true as far as I know